jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/java/JavaSamProjectWizard.kt

```diff
@@ -73,4 +82,24 @@ class SamHelloWorldGradle : JavaSamProjectTemplate() {
     override fun getDescription() = message("sam.init.template.hello_world.description")
 
     override fun dependencyManager(): String? = "gradle"
+
+    override fun postCreationAction(settings: SamNewProjectSettings, contentRoot: VirtualFile, rootModel: ModifiableRootModel) {
+        super.postCreationAction(settings, contentRoot, rootModel)
+        val buildFile = locateBuildFile(contentRoot, "build.gradle") ?: return
+
+        val gradleProjectSettings = GradleProjectSettings().apply {
+            withQualifiedModuleNames()
+            externalProjectPath = buildFile.path
+        }
+
+        val externalSystemSettings = ExternalSystemApiUtil.getSettings(rootModel.project, GradleConstants.SYSTEM_ID)
+        externalSystemSettings.setLinkedProjectsSettings(setOf(gradleProjectSettings))
+
+        val importSpecBuilder = ImportSpecBuilder(rootModel.project, GradleConstants.SYSTEM_ID)
+            .forceWhenUptodate()
+            .useDefaultCallback()
+            .use(ProgressExecutionMode.START_IN_FOREGROUND_ASYNC)
+
+        ExternalSystemUtil.refreshProjects(importSpecBuilder)
```

Comment:
Does it refresh on its own without this?

Reply:
I've manually tested. It doesn't auto refresh.